File: minimum-scalar/minimum-scalar.go

package main

import "fmt"
import "log"
import "bufio"
import "strings"
import "strconv"
import "os"

func main() {
  file, err := os.Open(os.Args[1])
  if err != nil {
    log.Fatal(err)
  }
  defer file.Close()

  scanner := bufio.NewScanner(file)
  var lines []string
  for scanner.Scan() {
    //'scanner.Text()' represents the test case, do something with it
    //fmt.Println(scanner.Text())
    //  lines.append(lines, scanner.Text())
    lines = append(lines, scanner.Text())
  }


  for i := 1; i < len(lines); i+=3 {
    first_array_s := strings.Split(lines[i + 1], " ")
    second_aray_s := strings.Split(lines[i + 2], " ")
    var first_array_i, second_array_i []int

    for i := 0; i < len(first_array_s); i++ {
      k, _ := strconv.Atoi(first_array_s[i])
      first_array_i = append(first_array_i, k)

      k, _ = strconv.Atoi(second_aray_s[i])
      second_array_i = append(second_array_i, k)
    }
    res :=

    for (var j = 0; j < first_aray.length; j++) {
      res += first_aray[j] * second_aray[j];
    }

    for i := 0; i < len(first_array_i); i++ {

    }
    fmt.Println(first_array_i)
    fmt.Println(second_array_i)

  }

  //fmt.Println(first_aray)
}

File: minimum-scalar/js/minimum-scalar.js
var fs  = require("fs");
var text = fs.readFileSync(process.argv[2]).toString().split('\n');

var testcases = text[0];

var stream = fs.createWriteStream("my_file.txt");
stream.once('open', function(fd) {
  for (var i = 1; i < text.length - 1; i+= 3) {
    first_aray = text[i + 1].split(' ').sort((function(a, b){return a-b}));;
    second_aray = text[i + 2].split(' ').sort((function(a, b){return b-a}));;

    var res = 0;
    for (var j = 0; j < first_aray.length; j++) {
      res += first_aray[j] * second_aray[j];
    }

  stream.write("Case #" + parseInt((i/3)+ 1) + ": " +res+"\n");
    //console.log("Case #" + parseInt((i/3)+ 1) + ": " +res);

  }


  stream.end();


});
